### ISSUE TITLE
Performance issues discussion

### ISSUE THREAD

**Issue by CShepartd** [mj1856/TimeZoneConverter#15]
I don't know why you closed issue #14 without any fix. Do you really think this is "hug-zone"? No, its ISSUES section and over 13000% slower than BCL and over 54000% slower than NodaTime is performace ISSUE

**Comment by mj1856** [comment]
It does not need to be a "hug-zone" as you put it, but it does need to reflect respectful discourse.  I didn't *have* to author this library.  I chose to.  You are free to not use it if you do not want to.  Yelling and being rude is not a way to act on *any* professional interaction.

To the issue at hand - Your tests do not demonstrate anything about TimeZoneConverter, they demonstrate primarily that `TimeZoneInfo.ConvertTime`, when given unique values via `DateTime.UtcNow`, has some overhead that is non-cachable (because the offset for each unique point in time may vary).

Also, in general, your approach of running the same task 1,000,000 times in a tight loop is called "micro-optimization", and while can show some interesting results, is generally considered non-constructive.  Most real world applications do not do such things.

For example, while you are showing this code:

```csharp
for (int i = 0; i < 1000000; i++)
{
    var a = TimeZoneInfo.ConvertTime(DateTime.UtcNow, TZConvert.GetTimeZoneInfo("Europe/Warsaw"));
}
```

I would expect a real-world application to do this instead:
```csharp
var tz = TZConvert.GetTimeZoneInfo("Europe/Warsaw");
for (int i = 0; i < 1000000; i++)
{
    var a = TimeZoneInfo.ConvertTime(DateTime.UtcNow, tz);
}
```

... and if I was in a situation where I needed to go through millions of timestamps with potentially different time zones, and I found the performance non-acceptable, then I would take it upon myself to add a cache on the time zone lookup.

Yes - such a cache can be built into the library, and I will keep this issue open to track this until I or someone else adds one.  You're welcome to send a PR if you like.
